docs/secret_derivation.md

```diff
@@ -0,0 +1,24 @@
+# Secret Derivation
+
+`fedimint-client` needs a `DerivableSecret`. Since `fedimint-client` is intended to be used as a library, the consumer of `fedimint-client` is free to arrive at this `DerivableSecret` in a number of ways. Here we describe a recommended approach to encourage interoperability.
+
+Let's say that the consuming application of `fedimint-client` has some `global_root_secret` that it may use for deriving all sorts of secrets deterministically. We then arrive at `fedimint-client`'s `DerivableSecret` using the derivation path:
+
+```
+global_root_secret/<key-type=per-federation=0>/<federation-id>/<wallet-number=0>/<key-type=fedimint-client=0>
+```
+
+For convenience, the provided `Bip39RootSecretStrategy` can be used to:
+- generate a BIP-39 mnemonic, and then
+- generate the `global_root_secret` from the mnemonic by calling `Bip39RootSecretStrategy::to_root_secret(mnemonic)`
+
+Then starting with the `global_root_secret`, the following derivation steps are performed:
+
+1. Call `global_root_secret.child_key(0)`. The 0 here indicates the first segment of the derivation path, `<key-type=per-federation=0>`. The returned value is `multi_federation_root_secret` of type `DerivableSecret`.
```

Comment:
Explaining remaining child ids might be useful. Do we want to reserve some namespace for internal usage and some for external (same for aux per-federation keys)?